RELEASE.md

```diff
@@ -1,66 +1,79 @@
 
-Due to either github dependencies or a crate failing the build stage during publish not all crates are being published.
+Whenever a `PATCH` is introduced, it is applied to all the latest `MAJOR` releases.
+For example: imagine there's releases `v1.0.0`, `v1.1.0`, and `v2.0.0`. A bug is found, dating back all the way to `v1.0.0`.
+A patch is applied such that the following new tags are introduced: `v1.1.1` and `v2.0.1`.
 
-- [x] buffer_sv2
-- [x] binary_codec_sv2
-- [x] binary_sv2
-- [x] common_messages_sv2
-- [x] const_sv2
-- [x] derive_codec_sv2
-- [x] framing_sv2
-- [x] serde_sv2
-- [x] sv1_api
-- [] noise_sv2 - github depependency (ed25519-dalek)
-- [] codec_sv2 - noise dependency
-- [] template_distribution_sv2 - build failure
-- [] mining_sv2 - publishes after template_distribution_sv2
-- [] job-declaration_sv2 - publishes after template_distribution_sv2
-- [] roles_logic_sv2 - publishes after template_distribution_sv2
-- [] network_helpers - publishes after template_distribution_sv2
-- [] error_handling - had to exclude to bypass rate limit but will probably publish now
-- [] sv2_ffi - had to exclude to bypass rate limit but will probably publish now
-- [] all roles - noise dependency
+# Git Branching Strategy
 
-## Github Roles Releases
+We follow a simplified [gitflow](https://nvie.com/posts/a-successful-git-branching-model/) branching strategy.
 
-The roles binaries are released in github in all jobs other than the `crates_publish` job. To be able to publish,
-the job must be able to find a new tag (one that hasnt already been released), matching the crate name of the role to be
-released. For example, imagine there is currently a release for pool_sv2-v1.0.0. For the jobs releasing a new pool version to succeed,
-it must be able to find a tag containing the string "pool_sv2" with a version greater than 1.0.0. This should not be an issue since the tags
-are automatically created by the `autoversion.yaml` workflow, and moved into main when the generated PR is merged.
+Although our strategy is very similar to the classic gitflow model, we do not keep release branches.
 
-### Github Release Issue
+![](git-branching.png)
 
-Currently we do not support windows releases because we are unable to run the step to return the last tag for a given crate name. See the section above
-for an explaination of the process.
+## Principal Branches
 
+The SRI repo holds two principal branches with an infinite lifetime:
+- `main`
+- `dev`
 
+We consider `main` to be the branch where the source code of `HEAD` always reflects a production-ready state.
 
-## Github Binary Releases
+We consider `dev` to be the branch where the source code of `HEAD` always reflects a state with the latest delivered development changes for the next release.
 
-# Versioning
+Every two weeks all the changes to `dev` are merged back into `main` and then tagged with a release number while bumping `MAJOR` and/or `MINOR`.
```

Comment:
```suggestion
Every two weeks all the changes to `dev` are merged back into `main` and then tagged with a release number while bumping `MAJOR` and/or `MINOR`.
```

After re-reading the PR the only thing I would be very careful is the pace of releases which we say is 2 weeks.
In startup/company structures this works, but my foss experience says we should probably be more conservative here no matter what type of release it is, I think even with 1-4 PR's it's obvious things can take quite a lot of time, especially if mainnet testing or any sort of testing is involved.

I would say we probablt shouldn't put a time-frame here, or we can put a 2-4weeks -  _until_ we do trial and error and see what kind of pace works for the structure/team we have here.

Trust me, 2-weeks is very tight in FOSS and can cause of lot of stress. As a reminder we're 8 months behind this release (ok it's major but just to give you a bit more realistic overview). What happens if we have PR's depending on each other and each of them requires testing (situation we had last 4-6 weeks).

FOSS structures are much loose and time-frames should adjust to that imo.

Reply:
I'm not opinionated on this, and I do see your point.

Over Discord (and maybe during today's call) I think you expressed a slightly different idea. There you're suggesting a 4 week release cycle.

Did you change your mind after writing this comment here?

Either way is fine, just asking so I can change the wording on the PR. Also, feel free to suggest some changes direct changes to `RELEASE.md` if you want.

Reply:
I would say 4 weeks was a consensus, but I still believe this is very optimistic. But for now we agreed we try with 4 weeks, and then see how it works for the team.